core/pta/attestation.c

```diff
@@ -0,0 +1,600 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2020, Elias von Däniken
+ */
+
+#include <compiler.h>
+#include <stdio.h>
+#include <kernel/pseudo_ta.h>
+#include <mm/tee_pager.h>
+#include <mm/tee_mm.h>
+#include <pta_attestation.h>
+#include <string.h>
+#include <string_ext.h>
+#include <malloc.h>
+#include <tee_api_types.h>
+#include <crypto/crypto.h>
+#include <crypto/crypto_impl.h>
+
+#define RSA_SIG_SIZE_BYTE 256
+
+/*
+ * Attestation data is static to keep until the device boots again.
+ * Afterwards the attestation data has to be reloaded and everthing
+ * has to be attestated newly.
+ */
+static struct attestation_data {
+	uint8_t *iv;
+	uint32_t iv_size;
+	uint8_t *pub;
+	uint32_t pub_size;
+	uint8_t *enc_priv;
+	uint32_t enc_priv_size;
+	uint8_t *sig;
+	uint32_t sig_size;
+} att_data;
+
+static TEE_Result priv_asn1_decode(uint8_t *priv,
+				uint8_t **d, size_t *d_size)
+{
+
+	uint32_t size;
```

Comment:
`= 0;` (coding rule, all local variables must be initialized)

Reply:
Fix [here](https://github.com/OP-TEE/optee_os/pull/4011/commits/63c627e3d7c547c5c5aaa71ef8c77817e9bbc304)